packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.ts

```diff
@@ -0,0 +1,155 @@
+import { GraphQLRequestClient } from '@sitecore-jss/sitecore-jss';
+
+export type GraphQLSitemapServiceConfig = {
+  endpoint: string;
+};
+
+type StaticPath = {
+  params: {
+    path: string[];
+  };
+};
+
+type SearchResult = {
+  search: {
+    results: { url: { path: string } }[];
+  };
+};
+
+export class GraphQLSitemapService {
+  constructor(private config: GraphQLSitemapServiceConfig) {}
+
+  /**
+   * Returns new graphql client instance
+   */
+  createClient(): GraphQLRequestClient {
+    const { endpoint } = this.config;
+
+    return new GraphQLRequestClient(endpoint);
+  }
+
+  /**
+   * Returns query to request root item id
+   * @param {string} rootItemPath root item path
+   */
+  getRootItemIdQuery(rootItemPath: string): string {
+    return `query {
+      item(path:"${rootItemPath}") {
+        id
+      }
+    }`;
+  }
+
+  /**
+   * Request root item id using path
+   * @param {string} rootItemPath root item path
+   */
+  async getRootItemId(rootItemPath: string): Promise<string | undefined> {
+    const query = this.getRootItemIdQuery(rootItemPath);
+
+    const data = await this.createClient()
+      .request<{ item: { id: string } }>(query)
+      .catch((error) => {
+        console.error('Error occured during fetching root item id:', error.response);
+
+        return null;
+      });
+
+    return data?.item.id;
+  }
+
+  /**
+   * Graphql query in order to fetch sitemap
+   * @param {string} rootItemId
+   * @param {locale} locale
+   */
+  getSitemapQuery(rootItemId: string, locale: string): string {
```

Comment:
Is there a way to override this query, or make it more flexible? What if I want to limit the paths that are generated at build time, because I am using `fallback: true` ? e.g. What if I wanted to exclude my large product catalog? Would be nice if a custom query could be provided w/out needing to recreate all the other logic/invocation around it.

Reply:
@nickwesselman Added ability to customize search query part. But I don't provide way to customize return type